python/ray/dashboard/dashboard.py

```diff
@@ -701,6 +718,106 @@ def run(self):
             if counter % 10:
                 self.update_nodes()
 
+class TuneCollector(threading.Thread):
+    """Initialize collector worker thread.
+    
+    Args
+        logdir (str): Directory path to save the status information of
+                        jobs and trials.
+        reload_interval (float): Interval(in ms) of space between loading 
+                        data from logs
+    """
+
+    def __init__(self, logdir, reload_interval):
+        super().__init__()
+        self._logdir = logdir
+        self._trial_records = {}
+        self._data_lock = threading.Lock()
+        self._reload_interval = reload_interval
+        self._available = False
+
+    def get_stats(self):
+        with self._data_lock:
+            return {
+                "trial_records": copy.deepcopy(self._trial_records)
+            }
+
+    def get_availability(self):
+        with self._data_lock:
+            return {
+                "available": self._available
+            }
+
+    def run(self):
+        while True:
+            with self._data_lock:
+                self.collect()
+            time.sleep(self._reload_interval)
+
+    def collect(self):
+        """
+        Collects and cleans data on the running Tune experiment from the Tune logs so that
+        users can see this information in the front-end client
+        """
+        sub_dirs = os.listdir(self._logdir)
+        job_names = filter(
+            lambda d: os.path.isdir(os.path.join(self._logdir, d)), sub_dirs)
+
+        self._trial_records = {}
+        for job_name in job_names:
+            analysis = Analysis(str(os.path.join(self._logdir, job_name)))
+            df = analysis.dataframe()
+            if len(df) == 0:
+                continue
+
+            self._available = True
+            df['trial_id'] = df['trial_id'].astype(str)
+            df = df.set_index("trial_id")
+            df = df.fillna(0)
+
+            j = df.to_dict(orient="index")
+            if len(j) > 0:
+                j = self.clean_trials(j, job_name)
+                self._trial_records.update(j)
+
+    def clean_trials(self, j, job_name):
+        first_trial = j[list(j.keys())[0]]
+        config_keys = []
+        float_keys = []
+        metric_keys = []
+        default_names = ["logdir", "time_this_iter_s", "done",  "episodes_total", "training_iteration", "timestamp", "timesteps_total",
+                                        "experiment_id", "date", "timestamp", "time_total_s", "pid", "hostname", "node_ip",
+                                        "time_since_restore", "timesteps_since_restore",  "iterations_since_restore", "experiment_tag"]
```

Comment:
There are probably linting issues here due to line length, be sure to run the Python linter before submitting.